### ISSUE TITLE
Lizzy create won't delete old stacks after rerouting traffic

### ISSUE THREAD

**Issue by priesus** [zalando/lizzy#178]
For our automated deployments on the test environment we use this command:
`lizzy create --keep-stacks 0 --traffic 100 senza.yaml ...`

Unfortunately it doesn't delete the old stack. Down below is the console output.

```
+ /tools/run :lizzy -e LIZZY_URL -- lizzy create --keep-stacks 0 --traffic 100 bam-release.yaml 201609150852lizzy2 201609150852
INFO: using registry.opensource.zalan.do/ci/toolchain-lizzy:cd24 as toolchain
Fetching authentication token.. . OK
WARNING: Artifact checking is still not supported by lizzy-client.
Requesting new stack.. OK
Stack ID: buying-article-management-release-201609150852lizzy2
Waiting for new stack... . . . . . . . . . . . . . . . . . . . . . . . . . . . . OK
Deployment Successful
Requesting traffic change.. OK
Deleting old stacks..[AGENT] Generating Cloud Formation template..You are deploying an internet-facing ELB that will be publicly accessible! You should have OAUTH2 and HTTPS in place!
[AGENT]  OK
[AGENT] Creating Cloud Formation stack buying-article-management-release-201609150852lizzy2.. OK

 buying-article-management-release-201609150852
[AGENT] Deleting Cloud Formation stack buying-article-management-release-201609150852.. EXCEPTION OCCURRED: An error occurred (ValidationError) when calling the DeleteStack operation: Stack [arn:aws:cloudformation:eu-central-1:759141842153:stack/buying-article-management-release-201609150852/c253d300-7b12-11e6-a042-50a68ae714c6] cannot be deleted while in status UPDATE_IN_PROGRESS
[AGENT] Validation Error: Stack [arn:aws:cloudformation:eu-central-1:<our_account_id>:stack/buying-article-management-release-201609150852/<some_uid_stuff>] cannot be deleted while in status UPDATE_IN_PROGRESS
```

BTW: The Jenkins build still succeeds. You actually have to look at the console output to see that there's an old stack left running.

I noticed similar behavior when using `senza` locally. After running `senza traffic`, both the old and the new stack are in the status UPDATE_IN_PROGRESS and you cannot delete them in AWS. It takes about a minute for the update to be finished; then you can delete the old stack.

Expected behavior:
When using `lizzy create` with both the `--traffic` and `--keep-stacks` options present, traffic is rerouted and old instances are deleted.

**Comment by priesus** [comment]
Ah, it probably only occurs when using `--keep-stacks 0`, because the instance to be deleted is affected by the traffic change.

**Comment by kaygorodov** [comment]
Hi,
I have the same error. I also use --keep-stacks 0 and --traffic 100, and it fails when it tries to remove the previous stack. The reason seems to be related to the traffic switching because it causes updating state of the previous stack.

```
[AGENT] Deleting Cloud Formation stack xxxx.. EXCEPTION OCCURRED: An error occurred (ValidationError) when calling the DeleteStack operation: Stack [xxxx] cannot be deleted while in status UPDATE_IN_PROGRESS
[AGENT] Validation Error: Stack [xxxx] cannot be deleted while in status UPDATE_IN_PROGRESS
```

**Comment by fogfish** [comment]
Hello,

This issue becomes a critical for out team. 

Do you have any ideas about work around?
Do you need any help to resolve the issue?

**Comment by priesus** [comment]
We implemented a workaround for this issue:
1. Deploy with /tools/run :lizzy -e LIZZY_URL -- lizzy create **--keep-stacks 1** --traffic 100 ...
2. Have a script run in intervals that removes unused stacks with `senza delete`

**Comment by rafaelcaricio** [comment]
Fixed by #82 released in version [2.5](https://github.com/zalando/lizzy/releases/tag/2.5).

**Comment by rafaelcaricio** [comment]
Actually this is going to be solved by https://github.com/zalando/lizzy-client/pull/82, sorry.
